app/controllers/assessment_questions_controller.rb

```diff
@@ -25,24 +27,26 @@ def choose_evidence_post
       @errors[:evidence_type] = 'You must choose a piece of evidence'
       render("assessments/choose-evidence") && return
     end
-    id = if params[:id] == "new"
-           SecureRandom.uuid
-         else
-           params[:id]
-         end
+
+    if params[:evidence_id] == "new"
+      evidence = Evidence.new
+      assessment = find_assessment
+      assessment.add_evidence(evidence)
+      save(assessment)
+    else
+      evidence = find_evidence
+    end
+
     if params[:choose_evidence] == "other"
       params[:evidence_type] = nil
     else
       params[:evidence_type_other] = nil
```

Comment:
As an aside, I think if you call `params[:non_existent_key] ruby will return `nil` anyway